src/library_webgpu.js

```diff
@@ -1718,12 +1825,13 @@ var LibraryWebGPU = {
     return data;
   },
 
+  // In webgpu.h offset and size are passed in as size_t.
+  // And we assume that size_t is always 32bit in emscripten.
   wgpuBufferGetMappedRange: function(bufferId, offset, size) {
     var bufferWrapper = WebGPU.mgrBuffer.objects[bufferId];
     {{{ gpu.makeCheckDefined('bufferWrapper') }}}
 
-    // TODO: if the sentinel value becomes WGPU_WHOLE_SIZE instead of 0, update this.
-    if (size === 0) size = undefined;
+    if (size === {{{ gpu.WHOLE_MAP_SIZE }}}) size = undefined;
```

Comment:
Emscripten 32bit question:
We have `size` passed in as `size_t` (in webgpu headers). How can I get it as u32 in `library_webgpu.js`?

The `makeGetU32` needs a ptr but `size` is passed in as a value in
`wgpuBufferGetMappedRange: function(bufferId, offset, size)`
For now js interprets it as -1 instead of 0xFFFFFFFF

If use `{{{ defineI64Param('size') }}}`. `size_low` and `size_high` will just become -1 and undefined (I think it's because essentially `size_t` is treated as 32 bit)

I just set `gpu.WHOLE_MAP_SIZE = -1` for now, but not sure if this is the right thing to do.

Reply:
Use `>>> 0` instead